### PR TITLE
Implement get_lp_token_price view instruction

### PR DESCRIPTION
**Pull request by adrena-orex**
Add `get_lp_token_price` view instruction allowing UI to access lp token price atomically.

include nit about some `u64` to `f64` conversions.

### CODE REVIEW

File: programs/perpetuals/src/instructions/get_lp_token_price.rs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+    let lp_supply = ctx.accounts.lp_token_mint.supply;
+
+    msg!("lp_supply: {}", lp_supply);
+
+    if lp_supply.is_zero() {
+        return Ok(0);
+    }
+
+    let ui_price_usd = checked_float_div(checked_as_f64(aum_usd)?, checked_as_f64(lp_supply)?)?;
```
Comment (askibin) [inline programs/perpetuals/src/instructions/get_lp_token_price.rs:65]: VM doesn't support floating point operations, they are emulated and hence slow. Also less precise for large integers comparing to u128. And you need to account for decimals in lp_supply. I think this should work:
let price_usd = math::checked_decimal_div(
                aum_usd as u128, -(Perpetuals::USD_DECIMALS as i32),
                lp_supply as u128, -(Perpetuals::LP_DECIMALS as i32),
                -(Perpetuals::USD_DECIMALS as i32),
            )?)?;
this would be the price you return, no need to do to_token_amount()

Comment (adrena-orex) [reply programs/perpetuals/src/instructions/get_lp_token_price.rs]: > VM doesn't support floating point operations, they are emulated and hence slow.

Interesting I didn't know that, thx 🙏 

Updated the instruction to use `math::checked_decimal_div`.

So I could understand `math::checked_decimal_div` edge cases better I've added some unit tests in `math` file:

```
 *  Executing task: cargo test --package perpetuals --lib -- math::tests --nocapture 

warning: profiles for the non root package will be ignored, specify profiles at the workspace root:
package:   /Users/orex/work/perpetuals/programs/perpetuals/Cargo.toml
workspace: /Users/orex/work/perpetuals/Cargo.toml
    Finished test [unoptimized + debuginfo] target(s) in 0.25s
     Running unittests src/lib.rs (target/debug/deps/perpetuals-930e8997802f4402)

running 1 test
test math::tests::test_checked_decimal_div_ok ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 13 filtered out; finished in 0.00s

 *  Terminal will be reused by tasks, press any key to close it.
 ```

Comment (Adrena-Corto) [reply programs/perpetuals/src/instructions/get_lp_token_price.rs]: Ha I know that I'm stupid I missed it. 

We used to use a fixed point math lib in the past and that made us stupid. Thanks

**Review by askibin — APPROVED** [review]
